lib/bdev/malloc/blockdev_malloc.c

```diff
@@ -126,7 +126,7 @@ blockdev_malloc_read(struct malloc_disk *mdisk, struct copy_task *copy_req,
 	SPDK_TRACELOG(SPDK_TRACE_MALLOC, "read %lu bytes from offset %#lx to %p\n",
 		      nbytes, offset, buf);
 
-	return spdk_copy_submit(copy_req, buf, mdisk->malloc_buf + offset,
+	return spdk_copy_submit(copy_req, buf, (void *)((char *)mdisk->malloc_buf + offset),
```

Comment:
Since gcc supported this case, I agree with your idea. Thanks you!

 I just reverted the  change of blockdev_malloc.c, please take a look.